### PR TITLE
Add with-pam-group meson option

### PR DESCRIPTION
**Pull request by terencode**

### CODE REVIEW

**Comment by terencode** [comment]
Should I update the ArchLinux packaging example to include the file?

**Comment by kakra** [comment]
You should first edit the meson file so this file gets installed. No need to put this into the packaging example yet because the the handling is not final.

**Comment by kakra** [comment]
@terencode If you're unsure how to handle the next steps, I'd volunteer to pick up your PR and rework it into a proper solution that also can be easily used by downstream packagers.

**Comment by terencode** [comment]
@kakra Sorry I forgot about it, I just updated it with what I hope is how you do it.

**Comment by kakra** [comment]
Now it would be good to put a config option for meson to actually turn this on, and additionally supply the group name. This would ease up acceptance for downstreams.

I'm not sure what the policy is here but I should make the commit summary shorter, you could instead add a description below it. And maybe you want to add a Signed-off-by line:

```
Short commit summary

Some few lines of optional description, or even some
elaborate description.

Signed-off-by: Full Name <email@domain.com>
```

**Comment by terencode** [comment]
@kakra please tell me what you think now.

**Comment by terencode** [comment]
Any feedback?

**Comment by kakra** [comment]
@terencode It seems Feral devs are busy right now... Maybe wait for the Linux SOTTR port be done? :-)

**Comment by aejsmith** [comment]
I think it'd be better to not have this installed by default, as AFAIK most distros do not use the "users" group by default.

You could change the README where it mentions that you need to configure limits.conf to say that the build option can be used to do this automatically for the specified group.

**Comment by terencode** [comment]
@aejsmith Thanks for the feedback, I modified the commit and the README to reflect your suggestion.

File: meson.build [hunk 1]
```diff
@@ -55,6 +55,19 @@ if with_systemd == true
     endif
 endif
 
+with_limits_conf = get_option('with-limits_conf')
```
[the remark this answers is not in the document]
Comment (aejsmith) [reply meson.build]: Yes, that's a better name I think. However I don't think it should always be installed - we're not creating the gamemode group so installing it with that won't be any use. I think for users installing themselves, they will likely want to use a different group (e.g. their own), and packagers will need to handle creating a group anyway so then they can opt in to installing the configuration.

[the remark this answers is not in the document]
Comment (terencode) [reply meson.build]: Renamed `--with-limits_conf` to `--with-pam-group` and keeping it as optional.

**Comment by kakra** [comment]
I think the SOB tagline should read `Signed-off-by` not `Signed-off by`.

**Comment by terencode** [comment]
@kakra Done, I had configured the global options but didn't know you could automate SOB (never did it before).
However, what's the point of signing only one commit? And why bother with appending the SOB (@aejsmith doesn't do it for example) ?

**Comment by kakra** [comment]
@terencode This is mostly to reflect your authorship/responsibility for this commit. If the patch would be modified by the maintainer during the merge, she or he can also add his SOB line and thus your credit will be retained.

**Comment by terencode** [comment]
@kakra a bit overkill but done anyway :wink: 
I now understand the SOB line thanks to your detailed explanation but still not for actually signing it with gpg...
What do you think about the PR now @aejsmith ?

**Comment by kakra** [comment]
@terencode BTW: I never forced you to use the SOB line but wrote "maybe you want to add" it. ;-)

**Comment by aejsmith** [comment]
Merged, thanks!

**Comment by terencode** [comment]
@kakra all good ;)

**Pull request merged**